backend/btrixcloud/basecrawls.py

```diff
@@ -496,13 +501,56 @@ async def delete_crawls_all_types(
         self, delete_list: DeleteCrawlList, org: Organization
     ):
         """Delete uploaded crawls"""
-        deleted_count, _, _, quota_reached = await self.delete_crawls(org, delete_list)
+        if len(delete_list.crawl_ids) == 0:
+            raise HTTPException(status_code=400, detail="nothing_to_delete")
+
+        deleted_count = 0
+
+        crawls_to_delete = await self._filter_delete_list_by_type(
```

Comment:
This all works, but there's a more optimal way without having to do the filtering by type here actually.
Since the delete function already calls get_crawl_raw, we have the type available there, and it can just bin the crawls by type there.

Reply:
See other comment - now splitting in one pass, but for now I think there are advantages to keeping `delete_crawls` to one type at a time.